src/components/Panel.ts

```diff
@@ -1,62 +1,61 @@
+/**
+ * Copyright (c) 2015 NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+
 import Viewport from "./Viewport";
 import { OriginalStyle, FlickingPanel, ElementLike } from "../types";
 import { DEFAULT_PANEL_CSS, EVENTS } from "../consts";
 import { addClass, applyCSS, parseArithmeticExpression, parseElement, getProgress } from "../utils";
 
 class Panel implements FlickingPanel {
+  public viewport: Viewport;
   public prevSibling: Panel | null;
   public nextSibling: Panel | null;
 
-  private element: HTMLElement;
-  private viewport: Viewport;
-  private state: {
+  protected state: {
     index: number;
     position: number;
     relativeAnchorPosition: number;
     size: number;
     isClone: boolean;
+    isVirtual: boolean;
     // Index of cloned panel, zero-based integer(original: -1, cloned: [0, 1, 2, ...])
     // if cloneIndex is 0, that means it's first cloned panel of original panel
     cloneIndex: number;
     originalStyle: OriginalStyle;
-    clonedPanels: Panel[];
     cachedBbox: ClientRect | null;
   };
+  private element: HTMLElement;
   private original?: Panel;
+  private clonedPanels: Panel[];
 
   public constructor(
     element: HTMLElement,
     index: number,
     viewport: Viewport,
   ) {
-    this.element = element;
     this.viewport = viewport;
     this.prevSibling = null;
     this.nextSibling = null;
+    this.clonedPanels = [];
 
     this.state = {
       index,
       position: 0,
       relativeAnchorPosition: 0,
       size: 0,
-      clonedPanels: [],
       isClone: false,
+      isVirtual: false,
       cloneIndex: -1,
       originalStyle: {
-        className: element.getAttribute("class") || null,
-        style: element.getAttribute("style") || null,
+        className: element ? element.getAttribute("class") : null,
+        style: element ? element.getAttribute("style") : null,
```

Comment:
@WoodNeck Could element be null?

Reply:
Oh, I forgot to get rid of that thing. I think it should be
```js
className: element.getAttribute("class")
```
I'll fix it right away. Thanks.